service/dap/server.go

```diff
@@ -486,19 +500,11 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 		s.binaryToRemove = debugname
 	}
 
-	// TODO(polina): support "remote" mode
-	if mode != "exec" && mode != "debug" && mode != "test" {
```

Comment:
Good idea to move this check. It did accidentally got pushed to a suboptimal spot as the function evolved. Thank you for catching.